public/js/src/interface.js

```diff
@@ -0,0 +1,45 @@
+showHeadlinesAndPictures();
+resetHashUrl();
+showSummaryOnClick();
+
+function showHeadlinesAndPictures() {
+  setTimeout(() => {
+    let html = ''
+    for (let i = 0; i < frontPage.articles.length; i ++) {
+      const headline = frontPage.articles[i].fields.headline
+      const thumbnail = frontPage.articles[i].fields.thumbnail
+      html += `<img class="image" src="${thumbnail}"><br>`
+      html += `<a class="headline" href="#${i}">${headline}</a><br>`
```

Comment:
Same comment here about violating encapsulation by reaching deep into the internals of `FrontPage`. 

You could make this functionality part of the Article and FrontPage classes, e.g. by adding `toHTML()` methods to each. You might also find that testing is a little easier because you can test the HTML output as part of testing those classes.

Reply:
Thanks, I definitely agree. I really wanted to extend the classes but ran out of time. Hopefully will find the time to go back and do this